File: padroesCriacionais/padrao-de-projeto-factory/java/src/test/java/PagarMeTest.java
import factory.MercadoPago;
import factory.PagarMe;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class PagarMeTest {
    public PagarMe pagarMe;

    @BeforeEach
    public void setUp() {
        this.pagarMe = new PagarMe();
    }

    @Test
    public void deveRealizarPedidoPixComSucesso() {
        String response = this.pagarMe.pix("dados do pedido", "dados do usuario");
        assertEquals("dados do pagamento com pix na Pagarme", response);
    }

    @Test
    public void deveRealizarPedidoBoletoComSucesso() {
        String response = this.pagarMe.boleto("dados do pedido", "dados do usuario");
        assertEquals("dados do pagamento com boleto na Pagarme", response);
    }

    @Test
    public void deveRealizarPedidoCreditoComSucesso() {
        String response = this.pagarMe.credito("dados do pedido", "dados do usuario");
        assertEquals("dados do pagamento com credito na Pagarme", response);
    }
}

File: padroesCriacionais/padrao-de-projeto-factory/php/tests/unit/PagarMeTest.php
<?php

namespace Factory\Test;

use Factory\Php\PagarMe;
use PHPUnit\Framework\TestCase;

class PagarMeTest extends TestCase
{
    private ?PagarMe $pagarMe = null;

    /**
     * @before
     */
    public function setUp(): void
    {
        $this->pagarMe = new PagarMe();
    }

    public function testDeveRealizarPedidoPixComSucesso()
    {
        $response = $this->pagarMe->pix("dados do pedido", "dados do usuario");
        $this->assertEquals("dados do pagamento com pix na Pagarme", $response);
    }

    public function testDeveRealizarPedidoBoletoComSucesso()
    {
        $response = $this->pagarMe->boleto("dados do pedido", "dados do usuario");
        $this->assertEquals("dados do pagamento com boleto na Pagarme", $response);
    }

    public function testDeveRealizarPedidoCreditoComSucesso()
    {
        $response = $this->pagarMe->credito("dados do pedido", "dados do usuario");
        $this->assertEquals("dados do pagamento com credito na Pagarme", $response);
    }
}